### DISCUSSION TITLE
How to exclude X86/X64 ABIs from Build/APK Bundle?

### DISCUSSION THREAD

**Discussion by shadowfoxish** [dotnet/maui#13058] [Q&A]
I was working on the publish aspect of my project to start getting the new APKs on devices, and I noticed the size of the bundle has burgeoned after the XF 5->.NET 7 change. A lot of it seems to be related to AOT files in the lib, but I'm okay with the advantages of that. The real culprit in size growth is the inclusion of x86 and x64 ABIs, which I don't need. 

![image](https://user-images.githubusercontent.com/12143535/216094080-0d7885b6-4a42-4910-9b13-788a2351df8c.png)

![image](https://user-images.githubusercontent.com/12143535/216094772-6620e18f-3503-4bce-a7d1-9532743ea480.png)

I did try playing around with the solution configuration, and removed Any CPU, and switched things to ARM64, but the x86 (and 32-bit ARM) keeps compiling. I think I'm looking in the wrong place. Any ideas?

I did find the "package per ABI" setting, which is close to what I'd be okay with, but I would like it to keep the 32/64 bit ARM builds together.

**Comment by junaidp27** [comment]
As far as I understand, you only want to include ABIs (architectures) that you are targeting. eg. you want to include ABI only for armeabi-v7a.

```xaml
<PropertyGroup>
    ...
    <AndroidSupportedAbis>armeabi-v7a</AndroidSupportedAbis>
    <RuntimeIdentifiers Condition="$(TargetFramework.Contains('-android'))">android-arm</RuntimeIdentifiers>
</PropertyGroup>
```

This way I was able to build archive just for armv7, and not the other ABIs.
